### PR TITLE
Merge improved 2080 code

### LINKED ISSUE #5: Scafolding for time track
A model for time track:

-response_map_id (int) This should be automatically created when ref to resposne_map

-time_data(string)

### LINKED ISSUE #7: Selectively merge previous code from 1989 and 2080
Merge needed files from previous implementations of the project which are understood by team and are to be documented for new implementation

### PR DESCRIPTION
**Pull request by boergerj**
Functionality:
- Tracks time user spends during a review on each file
- Displays entry per round in review report table for UI to be displayed with PI chart and time table

Changes:
- Pulled initial 2080 implementation
- After reading, understanding, and documenting implementation, made small reworks to improve merge with current beta branch
- Still lots of changes I want to make that I have documented but want to add initial working implementation and functionality for other branches to take advantage of

### CODE REVIEW

File: app/controllers/submission_viewing_events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    submission_viewing_event = LocalSubmittedContent.new(submission_viewing_event_params)
+    store.save(submission_viewing_event)
+
+    #if creating start time for expertiza update end times for all other links.
+    if param_args[:link]=='Expertiza Review' 
+      params[:submission_viewing_event][:link] = nil
+      params[:submission_viewing_event][:end_at] = params[:submission_viewing_event][:start_at]
+      record_end_time()
+    end
+    render nothing: true
```
Comment (AvlWx2014) [inline app/controllers/submission_viewing_events_controller.rb:34]: This is an example of methods from this implementation that I would love to simplify. It doesn't seem like as much code needs to be here as there is, and this is not the only method my sentiment applies to. This is not something I think needs to be done in order to merge this, but I thought I'd share.

File: app/helpers/submitted_content_helper.rb [hunk 2]
```diff
@@ -119,4 +121,120 @@ def self.unzip_file(file_name, unzip_dir, should_delete)
   # rescue
   # end
 end
+
+  #Holds information related to a link pressed during a review
+  class LocalSubmittedContent
```
Comment (AvlWx2014) [inline app/helpers/submitted_content_helper.rb:126]: I will keep and re-use this.

File: app/helpers/submitted_content_helper.rb [hunk 3]
```diff
@@ -119,4 +121,120 @@ def self.unzip_file(file_name, unzip_dir, should_delete)
...
+    def ==(other)
+      return @map_id = other.map_id && @round == other.round && @link == other.link &&
+      @start_at == other.start_at && @end_at == other.end_at &&
+      @created_at == other.created_at && @updated_at == other.updated_at
+    end
+
+  end
+
+  #LocalStorage implementation using PStore
+  class LocalStorage
```
Comment (AvlWx2014) [inline app/helpers/submitted_content_helper.rb:163]: This is going to be extracted to another file, and re-worked using redis.

**Review by AvlWx2014 — APPROVED** [review]

**Review by Hatsukikeika — APPROVED** [review]

File: app/controllers/submission_viewing_events_controller.rb (hunk 1 above)
Comment (boergerj) [reply app/controllers/submission_viewing_events_controller.rb:34]: Totally agree. On my list of things to do

**Pull request merged**